### ISSUE TITLE
Unable to add Unicorn ORDER to user

### ISSUE THREAD

**Issue by rojagit** [aws-samples/aws-serverless-appsync-loyalty#3]
Hitting the Order button does nothing, how can I debug ?

**Comment by KoldBrewEd** [comment]
Need more info than that. There should be an error in the console, start from there.

**Comment by rojagit** [comment]
Its devs who tinker around here; if there was an error in the console or anywhere I wudv'e reported it; re-doing the whole thing again, seems like the unicorns are not getting added for my user; nor is the app reporting any warnings

**Comment by KoldBrewEd** [comment]
I am talking about the browser developer console. You asked how you can debug, the console will give you debug information. Without an error I cannot help you.

Other than that, you should try to tinker. It's fun :)

A good way to test is to execute the query to create the order in the AppSync console and check if you get an error there.

I cannot replicate your issue, suggest to take a look at the live coding video for some insights on how to debug. Specially the part where I'm building and testing the mutations in the AppSync console.

**Comment by rojagit** [comment]
@awsed So now it works a bit better. After I re-did everything, I was able to see the unicorns AND I was able to order one YAY :)

Then disaster -  I logged out & back on to see it reflected, but this time ALL unicorns have disappeared from the app, though I can still see them in DD, but the balance is reflected correctly, but I dont see anything and obviously order button doesnt work anymore. Tried 3 diff. users each time with the hassle of a dozen MFA auths
![aws](https://user-images.githubusercontent.com/4193031/44540408-bb45ee00-a6d4-11e8-8df6-f0ed0cfc8a00.PNG)
![aws](https://user-images.githubusercontent.com/4193031/44540439-d153ae80-a6d4-11e8-94d9-32e6a680978b.PNG)

The only other thing I see each time is the radix error which I am assuming we can ignore?
./src/components/Unicorns.js
  Line 83:  Missing radix parameter  radix
Ok although I fixed that, I still encounter no unicorns in the list

**Comment by rojagit** [comment]
Debugger shows an error:

Debugger listening on ws://127.0.0.1:15454/d14c28de-056e-42a8-83cc-f2bf58e26602
For help see https://nodejs.org/en/docs/inspector
Debugger attached.
/home/ec2-user/environment/unicorn-loyalty/src/components/Unicorns.js:1
(function (exports, require, module, __filename, __dirname) { import React, { Component } from 'react';
                                                              ^^^^^^
(((( note that the hilite is wrongly placed on filename here but is under import as in the pic)))))
SyntaxError: Unexpected token import
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:616:28)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
Waiting for the debugger to disconnect...


Process exited with code: 1

![image](https://user-images.githubusercontent.com/4193031/44541004-b124ef00-a6d6-11e8-9ff4-0172c58fb5c6.png)

**Comment by rojagit** [comment]
Well I guess we can ignore that one as the IDE may not be geared for react, but in the console now I do see object retrieval is successful, everything is there, the UnicornList just not being rendered

(Do we need to call NPT...haah)

So between the first time everything appeared in full unicorn glory exactly as in your SS in the tut, and the 2nd time it disappeared on me, I did not change anything at all, all I did was logout and back.

**Comment by rojagit** [comment]
I see this: 
1. <td> cannot appear inside tbody - just a warning
2. VM828 bundle.js:109631 Uncaught (in promise) {data: {…}, errors: Array(2)}

Haha - now I see this "Download the React DevTools for a better development experience: https://fb.me/react-devtools" - will try that

**Comment by rojagit** [comment]
I see 8 Unicorns being retrieved (in log), and this empty list
![image](https://user-images.githubusercontent.com/4193031/44545186-60b38e80-a6e2-11e8-8752-2da1753f3efb.png)

**Comment by KoldBrewEd** [comment]
There'll be an error in the browser developer console in **red** if you are not seeing unicorns. Something like:

<img width="332" alt="graphql notes app 2018-08-24 15-12-53" src="https://user-images.githubusercontent.com/12226594/44566324-4ba53100-a7b0-11e8-9a99-90ba0d77aaad.png">

Alternatively run the same query the client is sending to AppSync in the AppSync Console:

```
query {
  listInventory{
    items{
        itemId
        itemDescription
        price
        count
    }
  }
}
```

Do you get an error in the AppSync Console?

**Comment by rojagit** [comment]
Like I said earlier (but I blame myself for too much blah), I do see the correct # of unicorns being retrieved in the browser log, they are just not being rendered by the react js script.
(I know about looking for the reds and pretty sure I saw all of them, 'data' DID have the unicorns in it.)

 I was going over you guys' awesome youtube vid when I realized I did not set 'Auto Generate GraphQL' to true, so thought that might be my mistake (left it at default FALSE as it was not mentioned in the steps here), so I started deleting my DSs in order to re-do them with that flag ON then fell into delete Resolver hell and I just gave up and am starting with a new account.

The disconnect here is your video is manually doing stuff (which is great) while this page tried to automate most of the stuff (which is good of coz) so I am still getting my head comparing the video and these scripts and trying to make out what might be missing.

So by the time I got your post I could not test that out to prove I was indeed seeing the Unics in the logs just not in the web page.

So here goes another attempt - hey I am learning stuff each time so it's all good.

**Comment by rojagit** [comment]
C9 & Amplify dont seem to be ready for prime time; While dropping files into c9 (Upload Local Files) it didnt give any feedback and I thought it was doing it multiple times , since I dropped files multiple times due to no feedback unlike the last time (took 30min) so I cancelled; then I tried again, dropping it just once, it took over 2 hours; then I logged off and when I came back on, my env in the IDE was gone!!

I had a similar experience with c9 in the past but thought it might be better with the AWS hookup.

What a terrible waste of time this has been ! Closing this as I am not interested in proceeding further anymore.

People on Stack & Reddit are right about the cloud - undependable.
